sources/whippet/src/dbus_policy.rs

```diff
@@ -0,0 +1,446 @@
+    pub(crate) bus_type: String,
+}
+
+/// Represents the actual policy in the dbus-launcher, as with the DbusPolicy, the order of the
+/// fields is crucial
+/// See:
+/// https://github.com/bus1/dbus-broker/blob/b0db0890d1254477cf832e5f9f0a798360c80fd9/src/launch/policy.c#L930
+#[derive(Debug, Type, Serialize, Clone, ZVariantValue, Default)]
+pub struct PolicyBatch {
+    pub(crate) connect_verdict: bool,
+    pub(crate) connect_priority: u64,
+    pub(crate) own_rules: Vec<OwnRecord>,
+    pub(crate) send_rules: Vec<SendReceiveRecord>,
+    pub(crate) recv_rules: Vec<SendReceiveRecord>,
+}
+
+/// Represents an Own Record in the actual dbus policy
+/// See:
+/// https://github.com/bus1/dbus-broker/blob/b0db0890d1254477cf832e5f9f0a798360c80fd9/src/launch/policy.c#L836
+#[derive(Debug, Type, Serialize, Clone, ZVariantValue, Default)]
+pub struct OwnRecord {
+    pub verdict: bool,
+    pub priority: u64,
+    pub prefix: bool,
+    pub name: String,
+}
+
+/// Represents a Send Record in the actual dbus policy
+/// See:
+/// https://github.com/bus1/dbus-broker/blob/b0db0890d1254477cf832e5f9f0a798360c80fd9/src/launch/policy.c#L877
+#[derive(Debug, Type, Serialize, Clone, ZVariantValue, Default)]
+pub struct SendReceiveRecord {
+    pub verdict: bool,
+    pub priority: u64,
+    pub name: String,
+    pub path: String,
+    pub interface: String,
+    pub member: String,
+    pub record_type: MessageType,
+    pub broadcast: u32,
+    pub min_fds: u64,
+    pub max_fds: u64,
+}
+
+impl DbusPolicy {
+    /// Builds a new DbusPolicy object using "system" as the only supported bus_type
+    fn new() -> Self {
+        Self {
+            uid_entries: Vec::new(),
+            console_entries: Vec::new(),
+            selinux_contexts: Vec::new(),
+            apparmor_enabled: false,
+            bus_type: String::from("system"),
+        }
+    }
+}
+
+impl TryFrom<Policy> for DbusPolicy {
+    type Error = crate::error::Error;
+
+    /// Convert TOML config to DbusPolicy format, inserting the default policy batch as the policy
+    /// for the default user, and appending the same batch to all the policy batches
+    /// See:
+    /// https://github.com/bus1/dbus-broker/blob/b0db0890d1254477cf832e5f9f0a798360c80fd9/src/launch/policy.c#L1056
+    fn try_from(policy: Policy) -> Result<Self> {
+        debug!("Converting TOML config to D-Bus policy format");
+
+        let mut dbus_policy = Self::new();
+
+        let default_batch = if let Some(default_policy) = &policy.default {
+            default_policy.try_into()?
+        } else {
+            // If there isn't a default policy batch, create one with verdict false and priority 1
+            // as the dbus-launcher does
+            PolicyBatch {
+                connect_verdict: false,
+                connect_priority: 1,
+                own_rules: Vec::new(),
+                send_rules: Vec::new(),
+                recv_rules: Vec::new(),
+            }
```

Comment:
nit: leverage the default impl to make it clear what's actually changing
```suggestion
            let policy_batch = Default::default();
            policy_batch.connect_priority = 1;
            policy_batch
```